main.go

```diff
@@ -33,56 +32,53 @@ func main() {
-		}
-		svcType := lbsvc.Spec.Type
-		if svcType != corev1.ServiceTypeLoadBalancer {
-			log.Fatalf("Expected LoadBalancer service type for service %s/%s but got %s.\n", lbsvc.Namespace, lbsvc.Name, svcType)
-		}
-		lbstatus := lbsvc.Status.LoadBalancer.Ingress
-		if len(lbstatus) != 1 {
-			log.Fatalf("Expected exactly one ip for load balancer but got %d.\n", len(lbstatus))
-		}
-		ip := lbstatus[0].IP
+	lbsvc, err := clientset.CoreV1().Services(svcNamespace).Get(svcName, metav1.GetOptions{})
+	if err != nil {
+		log.Fatalf("Could not get service %s/%s: %v\n", svcNamespace, svcName, err)
+	}
+	svcType := lbsvc.Spec.Type
+	if svcType != corev1.ServiceTypeLoadBalancer {
+		log.Fatalf("Expected LoadBalancer service type for service %s/%s but got %s.\n", lbsvc.Namespace, lbsvc.Name, svcType)
+	}
+	lbstatus := lbsvc.Status.LoadBalancer.Ingress
+	if len(lbstatus) != 1 {
+		log.Fatalf("Expected exactly one ip for load balancer but got %d.\n", len(lbstatus))
+	}
+	ip := lbstatus[0].IP
 
-		ingresses, err := clientset.ExtensionsV1beta1().Ingresses("").List(metav1.ListOptions{})
-		if err != nil {
-			log.Fatalf("Could not list ingresses: %v\n", err)
-		}
+	ingresses, err := clientset.ExtensionsV1beta1().Ingresses("").List(metav1.ListOptions{})
+	if err != nil {
+		log.Fatalf("Could not list ingresses: %v\n", err)
+	}
+
+	for _, ingress := range ingresses.Items {
+		bounce := false
+		for _, rule := range ingress.Spec.Rules {
+			host := rule.Host
+
+			if found, err := ipHost(host, ip); err != nil || !found {
+				log.Infof("%s does not resolve to %s.\n", host, ip)
+				continue
+			}
 
-		for _, ingress := range ingresses.Items {
-			bounce := false
-			for _, rule := range ingress.Spec.Rules {
-				host := rule.Host
-
-				if found, err := ipHost(host, ip); err != nil || !found {
-					log.Infof("%s does not resolve to %s.\n", host, ip)
-					continue
-				}
-
-				conn, err := tls.Dial("tcp", host+":443", nil)
-				if err != nil {
-					bounce = true
-					break
-				}
-				conn.Close()
+			conn, err := tls.Dial("tcp", host+":443", nil)
+			if err != nil {
+				bounce = true
+				break
 			}
-			if bounce {
-				err = bounceIngress(ingress, clientset)
-				if err != nil {
-					log.Errorf("Error bouncing ingress %s in namespace %s. Cluster may be in inconsistent state: %v\n", ingress.Name, ingress.Namespace, err)
-				} else {
-					log.Infof("Successfully bounced ingress %s in namespace %s.\n", ingress.Name, ingress.Namespace)
-				}
+			conn.Close()
+		}
+		if bounce {
+			err = bounceIngress(ingress, clientset)
+			if err != nil {
+				log.Errorf("Error bouncing ingress %s in namespace %s. Cluster may be in inconsistent state: %v\n", ingress.Name, ingress.Namespace, err)
+			} else {
+				log.Infof("Successfully bounced ingress %s in namespace %s.\n", ingress.Name, ingress.Namespace)
 			}
 		}
-
-		log.Infof("Sleeping at time %s for 5 minutes\n", time.Now().String())
-		time.Sleep(5 * time.Minute)
 	}
+
+	log.Infof("Bouncing completed 💯")
```

Comment:
If Unicode causes our logs to break, I'm going to be mad

Reply:
it wont